internal/runners/packages/packages.go

```diff
@@ -3,26 +3,44 @@ package packages
 import (
 	"strings"
 
+	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/locale"
 	"github.com/ActiveState/cli/internal/output"
 	"github.com/ActiveState/cli/internal/prompt"
 	"github.com/ActiveState/cli/pkg/cmdlets/auth"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
 	"github.com/ActiveState/cli/pkg/platform/model"
 	"github.com/ActiveState/cli/pkg/project"
 )
 
 const latestVersion = "latest"
 
-func executeAddUpdate(out output.Outputer, prompt prompt.Prompter, language, name, version string, operation model.Operation) error {
+func executeUpdate(out output.Outputer, prompt prompt.Prompter, language, name, version string, operation model.Operation) error {
 	// Use our own interpolation string since we don't want to assume our swagger schema will never change
 	var operationStr = "add"
 	if operation == model.OperationUpdated {
 		operationStr = "update"
+	} else if operation == model.OperationRemoved {
+		operationStr = "removed"
 	}
 
-	fail := auth.RequireAuthentication(locale.T("auth_required_activate"), out, prompt)
-	if fail != nil {
-		return fail.WithDescription("err_activate_auth_required")
+	isHeadless := false
+	if !authentication.Get().Authenticated() {
+		anonymousOk, fail := prompt.Confirm(locale.T("prompt_headless_anonymous"), true)
+		if fail != nil {
+			// TODO: Maybe ignore on interrupt?
+			return errs.Wrap(fail.ToError(), "Error prompting to proceed anonymously during headless commit.")
+		}
+		isHeadless = anonymousOk
+	}
+
+	// Note: User also lands here if answering No to the question about anonymous commit.
```

Comment:
This was not explicitly mentioned in the ACs.  Alternatively we could do nothing here and write an error message.